lib/ffmpeg/transcoderbase.py

```diff
@@ -128,9 +128,13 @@ def gen_input_opts(self):
       opts += f" -hwaccel {hw}"
     else:
       opts = "-init_hw_device {hwaccel}={hwdevice}"
+      if vars(self).get("dhwframes", None) is not None:
+        opts += " -extra_hw_frames {dhwframes}"
       opts += " -hwaccel_output_format {hwaccel}"
     if "hw" == self.mode:
       opts += " -hwaccel {hwaccel}"
+      if vars(self).get("dhwframes", None) is not None:
+        opts += " -extra_hw_frames {dhwframes}"
```

Comment:
These logical conditions will add the option to the command-line twice.  Is that necessary?